### PR TITLE
chore(DIA-629): release AREnableNewTermsAndConditions

### PR DESCRIPTION
**Pull request by iskounen**
### Description

This PR retires the `AREnableNewTermsAndConditions` flag.

See: artsy/eigen#10696

### PR Checklist (tick all before merging)

- [x] I have not stored any sensitive keys/values/information in `Echo.json5` and in CI, or I have not changed anything in that file.

### CODE REVIEW

**Review by carlos-motiro — APPROVED** [review]

**Review by araujobarret — APPROVED** [review]

File: Echo.json5 [hunk 1]
```diff
@@ -35,7 +35,7 @@
     { name: 'AREnableNewOnboardingFlow', value: true },
     { name: 'AREnableNewRequestPriceEstimateLogic', value: true },
     { name: 'AREnableNewSearch', value: true },
-    { name: 'AREnableNewTermsAndConditions', value: false },
+    { name: 'AREnableNewTermsAndConditions', value: true }, // 2024-09-03 removed artsy/eigen#10696
```
Comment (gkartalis) [inline Echo.json5:38]: 👀 why are you flipping this to true if you are retiring it and it was false before?

Comment (araujobarret) [reply Echo.json5:38]: I assumed that was supposed to be released a long time ago and got lost maybe?

Comment (iskounen) [reply Echo.json5:38]: @gkartalis so this feature was never actually released...? 😱

Comment (gkartalis) [reply Echo.json5:38]: I guess so since it is not flipped to true

Comment (iskounen) [reply Echo.json5]: @gkartalis thanks for catching this. I've reached out to our PM to let them know that we failed to launch this feature back when, and will launch it with this PR instead.

Comment (gkartalis) [reply Echo.json5]: sounds good

**Pull request merged**